22.md

```diff
@@ -0,0 +1,179 @@
+NIP-22
+======
+
+Comment
+-------
+
+`draft` `optional`
+
+A comment is a threading note always scoped to a root event or an `S`-tag.
+
+It uses `kind:1111` with plaintext `.content` (no HTML, Markdown, or other formatting).
+
+Comments MUST point to the root scope using uppercase tag names (e.g. `K`, `E`, `A` or `S`)
+and MUST point to the parent item with lowercase ones (e.g. `k`, `e`, `a` or `s`).
+
+```js
+{
+  kind: 1111,
+  content: '<comment>',
+  tags: [
+    // root scope: event addresses, event ids, or S-tags.
+    ["<A, E, S>", "<address, id or S-value>", "<relay or web page hint>", "<root event's pubkey, if an E tag>"],
+    // the root kind or type if r.
+    ["K", "<root kind>"],
+
+    // parent item: event addresses, event ids, or s-tags.
+    ["<a, e, s>", "<address, id or s-value>", "<relay or web page hint>", "<parent event's pubkey, if an e tag>"],
+    // parent item kind
+    ["k", "<parent comment kind>"]
+  ]
+  // other fields
+}
+```
+
+Tags `K` and `k` MUST be present to define the event kind of the root and the parent items.
+
+`S` and `s` tags create scopes for hashtags, geohashes, URLs, and other external identifiers. They accept the following values:
+
+| Type | `S`/`s` tag | `K`/`k` tag |
+|- | - | - |
+| URLs | "r:`<URL, normalized, no fragment>`" | "r:`<scheme-host, normalized>`" |
+| Hashtags | "t:`<topic, lowercase>`" | "#:t" |
+| Geohashes| "g:`<geohash, lowercase>`" | "#:g" |
+| Podcast Feeds | "i:podcast:guid:`<id, lowercase>`" | "i:podcast:guid" |
+| Podcast Episodes | "i:podcast:item:guid:`<id, lowercase>`" | "i:podcast:item:guid" |
+| Books | "i:isbn:`<id, hyphenated>`" | "i:isbn" |
+| Movies | "i:isan:`<id, lowercase>`" | "i:isan" |
+| Papers | "i:doi:`<id, lowercase>`" | "i:doi" |
+
```

Comment:
```suggestion
`q` tags SHOULD be used when citing/quoting events and addresses in the `.content`

```json
["q", <event-id> or <event-address>, <relay-url>, <pubkey-if-a-regular-event>]
```
```

Reply:
I don't get what you are saying `q` is not a retweet of `1111`. 

`q` is used to post a `kind:1` inside a `1111`. It's irrelevant if `1111` is a reply or not.

Reply:
Here's the example: 

```json
{
  kind: 1111,
  content: 'This post is awesome: nostr:nevent1...',
  tags: [
    // nip-94 file event id
    ["E", "80c48d992a38f9c445b943a9c9f1010b396676013443765750431a9004bdac05", "wss://example.relay"],
    // the root kind
    ["K", "1063"]

    // parent 
    ["e", "80c48d992a38f9c445b943a9c9f1010b396676013443765750431a9004bdac05", "wss://example.relay"],
    // the parent kind
    ["k", "1063"]

    ["q", "nevent1 id", "nevent1 relay hint", "nevent1 author"]
  ]
  // other fields
}
```

Reply:
I think the name "Quote Reposts" is throwing you off. The NIP is calling a repost anything that has a quote inside, which is the kind 1 view of the world. But it's not a repost for everybody else. It's just a post with a quote in it. 

But it does that strictly to count the "Quotes" of each event. 

According to NIP-18
> It also allows you to easily pull and count all of the quotes for a post.

So, if a post is quoted inside a `1111` it should have the `q` tag so that the counter of the post is correct.